odoo/tools/safe_eval.py

```diff
@@ -376,11 +836,21 @@ def safe_eval(expr, globals_dict=None, locals_dict=None, mode="eval", nocopy=Fal
     except ZeroDivisionError:
         raise
     except Exception as e:
-        raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
-def test_python_expr(expr, mode="eval"):
+        raise ValueError(f"{type(e)} : {e} while evaluating\n{expr}")
+
+    eval_locals_dict = {k: v for k, v in eval_locals_dict.items() if "__" not in k}
+    eval_globals_dict = {k: v for k, v in eval_globals_dict.items() if "__" not in k}
+    locals_dict.update(eval_locals_dict)
+    globals_dict.update(eval_globals_dict)
+
+    return ret
```

Comment:
this is a new feature, where is it used ?

Reply:
wdym ?

Reply:
nevermind, we returned before, forget that comment

Reply:
Ok :)